components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpointTest.java

```diff
@@ -250,11 +251,17 @@ public void testIssueAccessToken(String clientId, String authzHeader, Object par
         requestParams.put(OAuth.OAUTH_USERNAME, new String[]{USERNAME});
         requestParams.put(OAuth.OAUTH_PASSWORD, new String[]{"password"});
 
+        OAuthClientAuthnContext oAuthClientAuthnContext = new OAuthClientAuthnContext();
+        oAuthClientAuthnContext.setAuthenticated(true);
```

Comment:
Is directly setting the client authentication to true correct like this? I can see in the data provider there is a scenario we are sending an invalid authz header as well

Reply:
That won't affect the `isAuthenticated` property. Coz, the values of the `OAuthClientAuthnContext` is getting populated from this line https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/model/CarbonOAuthTokenRequest.java#L235

Reply:
Irrespective of the validness of the request, this `isAuthenticated` variable becomes `false`

Reply:
earlier we might not have hit that code block but now you're changing the code to always set the client authentication to true. This can lead to erroneous test results especially since the data provider is sending an authz header. Have you verified that this has no impact on the current intended test scenarios?

Reply:
I have run the tests and all were passing